src/casp/services/api/services/consensus_engine/strategies.py

```diff
@@ -0,0 +1,226 @@
+import json
+import typing as t
+
+import numpy as np
+from smart_open import open
+
+from casp.services import settings
+from casp.services.api import schemas
+from casp.services.api.clients.matching_client import MatchResult
+from casp.services.api.data_manager import CellOperationsDataManager
+from casp.services.db import models
+
+
+class ConsensusStrategy(t.Protocol):
+    def summarize(self, query_ids: t.List[str], knn_query: MatchResult) -> t.List[schemas.QueryCellAnnotationAbstract]:
+        """
+        Define the method signature for the consensus strategy, which is responsible for summarizing query neighbor
+
+        :param query_ids: List of query cell IDs from the kNN query.
+        :param knn_query: The result of the kNN query.
+
+        :return: A list of summarized query neighbor context annotations.
+        """
+        ...
+
+
+class ConsensusStrategyType:
```

Comment:
maybe make this explicitly extend enum
```suggestion
class ConsensusStrategyType(t.Enum):
```